opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -310,13 +305,12 @@ def __init__(
         name: str,
         attributes: types.Attributes = None,
         timestamp: Optional[int] = None,
+        limit: Optional[int] = _DEFAULT_OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT,
     ) -> None:
-        super().__init__(name, timestamp)
-        self._attributes = attributes
-
-    @property
-    def attributes(self) -> types.Attributes:
-        return self._attributes
+        EventBase.__init__(self, name, timestamp)
+        Attributed.__init__(
+            self, attributes, limit=limit, immutable=True, filtered=True
```

Comment:
This feels inconsistent. We are implementing immutability in `Resource` by [returning a copy of `attributes`](https://github.com/open-telemetry/opentelemetry-python/pull/1893/files#diff-d2c46fd8da8afdf67c6ad3dd42b0a243c5049f3c720d9bb3351f1d5522338b06R189).

Reply:
i agree, the handling of attributes is inconsistent currently. I'm hoping this PR will help with that. I don't know which way is the better way to handle it, but I'm open to suggestions